base.py

```diff
@@ -589,16 +576,104 @@ def get_columns(self, connection, table_name, schema=None, **kw):
 
             type_instance = col_type(**col_type_kw)
 
-            cdict = {
-                'name': self.normalize_name(colname),
+            sfkey = ('snowflake_cache', schema, table_name)
+            if info_cache.get(sfkey) is None:
+                info_cache[sfkey] = {}
+
+            info_cache[sfkey][colname] = {
                 'type': type_instance,
                 'nullable': is_nullable == 'YES',
                 'default': column_default,
                 'autoincrement': is_identity == 'YES',
-                'primary_key': column_map[colname],
             }
 
-            columns.append(cdict)
+    @reflection.cache
+    def get_columns(self, connection, table_name, schema=None, **kw):
```

Comment:
is there's any way to avoid/reduce the code duplication with _get_columns_for_tables? This is a complicated method and it may be easier to maintain if there was a single copy.

Reply:
sure, will do the refactoring later today.